### ISSUE TITLE
having difficulty in changing the total number of epoches

### ISSUE THREAD

**Issue by blind-ez** [ultralytics/ultralytics#2980]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

At first, I set the total number of epoches to 300, but around 120th epoch, I decided to decrease it to 150.

So I fixed the train script as follows and resumed the training:

**!yolo detect train data=dataset.yaml model=/content/drive/MyDrive/datasets/runs/detect/train/weights/best.pt epochs=150 resume=True**

However, the change doesn't seem to be applied. Following message comes out: 

![1](https://github.com/ultralytics/ultralytics/assets/129811445/b653990e-9ca1-4c0e-9c0a-b42c9d7fae78)

How can i successfully change the total number of epoches?

Thanks for your help.

### Additional

After I finish the total training of 150 epoches, I might retrain the pretrained model with lower learning rate for additional 50 epoches. Following script will work?

!yolo detect train data=dataset.yaml model=/content/drive/MyDrive/datasets/runs/detect/train/weights/best.pt epochs=50 imgsz=800 lr0=0.005 lrf=0.005

**Comment by AbdesselamFerdi** [comment]
@blind-ez when you start the YOLOv8 training, a file called 'args.yaml' will be created in the directory: ('yolov8/runs/detect/train/yolov8'). This file contains the arguments used in yolov8 training including the number of epochs (epochs). So, to change the number of epochs when resuming training, you should change the epochs value in the 'args.yaml' file and the command is: `!yolo detect train data=dataset.yaml model=/content/drive/MyDrive/datasets/runs/detect/train/weights/best.pt resume=True`
Hope this help you!

**Comment by blind-ez** [comment]
@AbdesselamFerdi Really appreciating your help, but how can I change the value in the 'args.yaml' file? because it is created 'after' I run the command which trains the model as you mentioned.

That is, before I run the command below, there is no 'args.yaml', is there?
!yolo detect train data=dataset.yaml model=/content/drive/MyDrive/datasets/runs/detect/train/weights/best.pt resume=True

**Comment by AbdesselamFerdi** [comment]
@blind-ez simply you go to the 'args.yaml' file in 'yolov8/runs/detect/train/yolov8' directory and change the epochs value manually.

**Comment by blind-ez** [comment]
@AbdesselamFerdi Oh, I manually changed it but it still outcomes the same message as I attached, And then if i reopen the arg file, the epoches value returns to 300 automatically.. haha..

![1](https://github.com/ultralytics/ultralytics/assets/129811445/eb207fb5-375d-41aa-b3e0-38ad15609916)

**Comment by AbdesselamFerdi** [comment]
@blind-ez I apologize for any confusion in the previous response.
You cannot resume training with additional epochs by modifying the `args.yaml` file. So, you should start a new training with the recently saved checkpoint file for the new number of epochs using the command: `!yolo detect train data=dataset.yaml model=/content/drive/MyDrive/datasets/runs/detect/train/weights/best.pt epochs=50` 
Hope this help you!

**Comment by glenn-jocher** [comment]
Hi @blind-ez,
 
If you have modified the 'args.yaml' file to change the number of epochs and still you're receiving the same error while resuming, it may be because the new settings are not being read properly or the file is being overwritten with the old value when resuming. Instead of modifying 'args.yaml', you can directly pass the `epochs` argument while resuming training using this command: `!yolo detect train data=dataset.yaml model=/content/drive/MyDrive/datasets/runs/detect/train/weights/best.pt epochs=150 resume=True`. 

Moreover, if you want to retrain the pretrained model with a lower learning rate for additional 50 epochs, you should run a new training with the newly saved checkpoint file and new settings using `!yolo detect train data=dataset.yaml model=/content/drive/MyDrive/datasets/runs/detect/train/weights/best.pt epochs=50 imgsz=800 lr0=0.005 lrf=0.005`. 

I hope this explanation helps!

**Comment by virgilynt** [comment]
Hi, can somebody help me, i just want to change the number of epochs to 50. This is my first time running the code

**Comment by glenn-jocher** [comment]
@virgilynt hi there! 😊 Sure, to change the number of epochs to 50, you just need to include the `epochs=50` argument in your training command, like so:

```bash
!yolo detect train data=your_dataset.yaml model=your_model_file.pt epochs=50
```

Just replace `your_dataset.yaml` and `your_model_file.pt` with your dataset configuration file and model file, respectively. That's it! Let us know if you need further assistance. Happy training! 🚀

**Comment by virgilynt** [comment]
where i can find my dataset.yaml and model file?

**Comment by virgilynt** [comment]
I already change the yaml file in data.yaml and yolov8n.pt but it didnt work. Why?

**Comment by virgilynt** [comment]
![Uploading Screenshot (218).png…]()

**Comment by virgilynt** [comment]
![Uploading Screenshot (218).png…]()

**Comment by glenn-jocher** [comment]
@virgilynt i'm sorry, but it seems like the image you tried to upload didn't go through properly. Could you please try uploading it again or describe the issue you're encountering in more detail? This way, I'll be able to provide you with better assistance. For changing the `yaml` file and model, you typically need to ensure the paths are correctly specified in your command, like so:

```bash
!yolo detect train data=your_path/data.yaml model=your_path/yolov8n.pt epochs=50
```

Make sure `your_path` is the correct path to your files. If it still doesn't work, sharing the error message or part of the code where the issue occurs would be really helpful! 😊
